cliquet/tests/test_resource.py

```diff
@@ -0,0 +1,327 @@
+        )
+
+        arguments = viewset.collection_arguments(MagicMock, 'get')
+        schema = arguments.pop('schema')
+        self.assertDictEqual(
+            arguments,
+            {
+                'cors_headers': sentinel.cors_headers,
+                'cors_origins': sentinel.cors_origins,
+                'error_handler': sentinel.error_handler,
+                'permission': 'readonly'
+            }
+        )
+
+    def test_default_arguments_are_used_for_record_arguments(self):
+        default_arguments = {
+            'cors_headers': sentinel.cors_headers,
+        }
+        default_record_arguments = {
+            'cors_origins': sentinel.record_cors_origins,
+        }
+
+        record_get_arguments = {
+            'error_handler': sentinel.error_handler
+        }
+
+        viewset = ViewSet(
+            default_arguments=default_arguments,
+            default_record_arguments=default_record_arguments,
+            record_get_arguments=record_get_arguments
+        )
+
+        arguments = viewset.record_arguments(MagicMock, 'get')
+        schema = arguments.pop('schema')
+
+        self.assertDictEqual(
+            arguments,
+            {
+                'cors_headers': sentinel.cors_headers,
+                'cors_origins': sentinel.record_cors_origins,
+                'error_handler': sentinel.error_handler,
+                'permission': 'readonly'
+            }
+        )
+
+    def test_class_parameters_overwrite_each_others(self):
+        # Some class parameters should overwrite each others.
+        # The more specifics should prevail over the more generics.
+        # Items annoted with a "<<" are the one that should prevail.
+        default_arguments = {
+            'cors_origins': sentinel.default_cors_origins,
+            'error_handler': sentinel.default_error_handler,
+            'cors_headers': sentinel.default_cors_headers,  # <<
+        }
+        default_record_arguments = {
+            'cors_origins': sentinel.default_record_cors_origin,
+            'error_handler': sentinel.default_record_error_handler,  # <<
+        }
+
+        record_get_arguments = {
+            'cors_origins': sentinel.record_get_cors_origin,  # <<
+        }
+
+        viewset = ViewSet(
+            default_arguments=default_arguments,
+            default_record_arguments=default_record_arguments,
+            record_get_arguments=record_get_arguments,
+        )
+
+        arguments = viewset.record_arguments(MagicMock, 'get')
+        schema = arguments.pop('schema')
+
+        self.assertDictEqual(
+            arguments,
+            {
+                'cors_headers': sentinel.default_cors_headers,
+                'error_handler': sentinel.default_record_error_handler,
+                'cors_origins': sentinel.record_get_cors_origin,
+                'permission': 'readonly'
+            }
+        )
+
+    def test_service_arguments_arent_inherited_by_record_arguments(self):
```

Comment:
You could group arguments tests together into a ``CorniceArgumentsTest``, and put the one they have in common inside ``setUp``